### PR TITLE
at: 3.1.23 -> 3.2.5

### PR DESCRIPTION
**Pull request by anthonyroussel**
###### Description of changes

https://salsa.debian.org/debian/at/-/blob/master/ChangeLog
https://salsa.debian.org/debian/at/-/compare/upstream%2F3.1.23...upstream%2F3.2.5

Replaced install.patch with substituteInPlace

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/system/at/default.nix [hunk 1]
```diff
@@ -5,22 +5,38 @@
...
   ];
 
+  postPatch = ''
+    # Remove chown commands and setuid bit
+    substituteInPlace Makefile.in \
+      --replace '$(INSTALL) -o root -g root' '$(INSTALL)' \
+      --replace '$(INSTALL) -d -o root -g root' '$(INSTALL) -d' \
+      --replace '$(INSTALL) -g root -o root' '$(INSTALL)' \
+      --replace '$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME)' '$(INSTALL)' \
+      --replace '$(INSTALL) -o root -g $(DAEMON_GROUPNAME)' '$(INSTALL)' \
```
Comment (pbsds) [inline pkgs/tools/system/at/default.nix:31]:
```suggestion
      --replace ' -o root ' ' ' \
      --replace ' -g root ' ' ' \
      --replace ' -o $(DAEMON_GROUPNAME) ' ' ' \
      --replace ' -g $(DAEMON_GROUPNAME) ' ' ' \
```

File: pkgs/tools/system/at/default.nix [hunk 2]
```diff
@@ -5,22 +5,38 @@
...
+      --replace '$(DESTDIR)$(etcdir)' "$out/etc" \
+      --replace '$(DESTDIR)$(ATJOB_DIR)' "$out/var/spool/atjobs" \
+      --replace '$(DESTDIR)$(ATSPOOL_DIR)' "$out/var/spool/atspool" \
+      --replace '$(DESTDIR)$(LFILE)' "$out/var/spool/atjobs/.SEQ" \
```
Comment (pbsds) [inline pkgs/tools/system/at/default.nix:35]: Is it possible to set the following vars, as e.g. environs?

* `DESTDIR` = `$out`
* `etcdir` =  `/etc`
* `ATJOB_DIR` = `/var/spool/atjobs`
* `ATSPOOL_DIR` = `/var/spool/atspool`
* `LFILE` = `/var/spool/atjobs/.SEQ`

**Review by pbsds — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/at/default.nix (hunk 1 above)
Comment (anthonyroussel) [reply pkgs/tools/system/at/default.nix]: Hello. I've done the changes, thanks for the review!

File: pkgs/tools/system/at/default.nix (hunk 2 above)
Comment (anthonyroussel) [reply pkgs/tools/system/at/default.nix:35]: I don't understand why do you want me to do that?

These variables are already set during the configure phase if I'm not mistaken: https://salsa.debian.org/debian/at/-/blob/upstream/3.2.5/configure.ac#L170.

**Comment by pbsds** [comment]
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] all depending packages build
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

Result of `nixpkgs-review pr 207172` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>at</li>
  </ul>
</details>

**Review by pbsds — APPROVED** [review]

**Pull request merged**